collector/container/konflux.Dockerfile

```
@@ -0,0 +1,196 @@
+           -DCOLLECTOR_APPEND_CID=${COLLECTOR_APPEND_CID} \
+           -DUSE_VALGRIND=${USE_VALGRIND} \
+           -DADDRESS_SANITIZER=${ADDRESS_SANITIZER} \
+           -DTRACE_SINSP_EVENTS=${TRACE_SINSP_EVENTS} && \
+    cmake --build ${CMAKE_BUILD_DIR} --target all -- -j "${NPROCS:-2}" && \
+    ctest -V --test-dir ${CMAKE_BUILD_DIR} && \
+    strip -v --strip-unneeded "${CMAKE_BUILD_DIR}/collector/collector"
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest AS support-packages-downloader
+
+WORKDIR /staging
+
+COPY kernel-modules/MODULE_VERSION kernel-modules/MODULE_VERSION
+COPY collector/container/scripts/download-support-package.sh download-support-package.sh
+
+RUN ./download-support-package.sh
+
+# Do NOT use follow_tag here, as we do not need or want collector to be rebuilt
+# with each drivers build (which may become very frequent)
+FROM brew.registry.redhat.io/rh-osbs/rhacs-drivers-build-rhel8:0.1.0 AS drivers-build
+# FROM registry-proxy.engineering.redhat.com/rh-osbs/rhacs-drivers-build-rhel8:0.1.0 AS drivers-build
+
+# TODO(ROX-20312): we can't pin image tag or digest because currently there's no mechanism to auto-update that.
+FROM registry.access.redhat.com/ubi8/ubi-minimal:latest AS unpacker
+
+RUN microdnf install -y unzip findutils
+WORKDIR /staging
+
+COPY --from=support-packages-downloader /staging/support-pkg.zip /staging/
+COPY kernel-modules/MODULE_VERSION MODULE_VERSION.txt
+# Creating this directory ensures the scratch build with dummy support-pkg.zip will not fail.
+RUN mkdir -p "/staging/kernel-modules/$(cat MODULE_VERSION.txt)"
+# First, unpack upstream support package, only on x86_64
+RUN if [[ "$(uname -m)" == x86_64 ]]; then unzip support-pkg.zip ; fi
+# Fail non-scratch build if there were no drivers matching the module version.
+RUN if [[ "$(uname -m)" == x86_64 && "$(ls -A /staging/kernel-modules/$(cat MODULE_VERSION.txt))" == "" && "$(unzip -Z1 support-pkg.zip)" != "dummy-support-pkg" ]] ; then \
+      >&2 echo "Did not find any kernel drivers for the module version $(cat MODULE_VERSION.txt) in the support package"; \
+      exit 1; \
+    fi
+
+# Next, import modules from downstream build, which take priority over upstream, on non-x86 architectures
+# TODO(ROX-13563): find a way to not have to separately pull in the support package and downstream-built drivers.
+COPY --from=drivers-build /kernel-modules /staging/downstream
+RUN if [[ "$(uname -m)" != x86_64 ]]; then \
+      cp -r /staging/downstream/. /staging/kernel-modules/ ; \
+    fi
+
+# Create destination for drivers.
+RUN mkdir /kernel-modules
+# Move files for the current version to /kernel-modules
+RUN find "/staging/kernel-modules/$(cat MODULE_VERSION.txt)/" -type f -exec mv -t /kernel-modules {} +
+# Fail the build if at the end there were no drivers matching the module version.
+RUN if [[ "$(ls -A /kernel-modules)" == "" && \
+        !("$(uname -m)" == x86_64 && "$(unzip -Z1 support-pkg.zip)" == "dummy-support-pkg") ]]; then \
```

Scratch builds don't exist in Konflux so this check becomes
```suggestion
RUN if [[ "$(ls -A /kernel-modules)" == "" ]]; then \
```

Note that now it partially duplicates the check done above for x86_64.

The difference is that the above check verifies whether the _support package_ arrived without the matching drivers version. In case of x86_64, RHEL drivers (i.e. excluding non-Red Hat OS-es) should also come from `brew.registry.redhat.io/rh-osbs/rhacs-drivers-build-rhel8:0.1.0` which can mask the problem of mismatching (or empty) support package if the above check was removed.

This check here is most meaningful for ppc64le and s390x where drivers come _only_ from `rhacs-drivers-build-rhel8`.

Both checks are needed but there could probably be more clear and concise way to express them.

---

Applied your suggestion and simplified in https://github.com/stackrox/collector/pull/1441/commits/90535d5917c5ef6ee3c4ad4ea9acbfa945b33985 .